test-nixpkgs-check-by-name.nix

```
@@ -0,0 +1,38 @@
+{
+  lib,
+  runCommand,
+  nixpkgs-check-by-name,
+  initNix,
+  nixpkgs,
+  nix,
+  nixVersions,
+}:
+let
+  nixDerivations = lib.filterAttrs (
+    name: value:
+    let
+      eval = builtins.tryEval value;
+    in
+    eval.success && lib.isDerivation eval.value
+  ) nixVersions;
+
+  self =
+    nix:
+    runCommand "test-nixpkgs-check-by-name-with-${nix.name}"
+      {
+        nativeBuildInputs = [
+          nixpkgs-check-by-name
+          nix
+        ];
+        nixpkgsPath = nixpkgs;
+
+        # Allow running against all other Nix versions.
+        passthru = lib.mapAttrs (name: value: self value) nixDerivations;
```

```suggestion
        # Allow running against all other Nix versions.
        passthru.nixVersions = lib.mapAttrs (name: value: self value) nixVersions;
```

`passthru` attributes should always be static, so that there's no namespace overlap with other resulting derivation attributes.

And the `tryEval` mapping above is not necessary :)

---

> And the tryEval mapping above is not necessary :)

It is if you want to run against everything in the nixVersions array. Otherwise:

```
       error: nix_2_10 has been removed
```

---

I don't think we should worry about running it for all Nix version attributes, instead limiting it to `nixVersions.{minimum,stable,latest,git}` should be good enough.

---

I've kept the mapping, as it's quite useful to me to be able to run against all Nix and Lix versions, but I have added a top-level value -- which runs in CI -- for the following:

1. Nix minimum
2. Nix latest
3. Nix git
4. Lix stable
5. Lix latest